### PR TITLE
Add view that lists all used libraries and their licenses

### LINKED ISSUE #155: List used libraries and their licenses

### PR DESCRIPTION
**Pull request by chrgernoe**
Fixes #155 

For some reason, the license for the [Apache Commons CSV](https://commons.apache.org/proper/commons-csv/) library is empty. It is under clarification in issue https://github.com/mikepenz/AboutLibraries/issues/827. Since the information are generated automatically during each build, it should not be a blocker to merge this issue.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -156,6 +162,15 @@ dependencies {
...
     // For introspection
     implementation "org.jetbrains.kotlin:kotlin-reflect:1.6.0"
+
+    /*
+     * Listing all used libraries
+     *
+     * https://github.com/mikepenz/AboutLibraries
+     */
+    implementation "com.mikepenz:aboutlibraries-core:${aboutlibrariesVersion}"
+    //implementation "com.mikepenz:aboutlibraries-compose:${aboutlibrariesVersion}"
```
Comment (paetz) [inline app/build.gradle:172]: Please do not commit commented-out code

File: app/src/main/java/com/yacgroup/yacguide/AboutActivity.kt [hunk 2]
```diff
@@ -82,4 +87,9 @@ class AboutActivity : BaseNavigationActivity() {
             setNegativeButton()
         }.show()
     }
+
+    private fun _showLibraries() {
+        val intent = Intent(this, AboutLibrariesActivity::class.java)
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/AboutActivity.kt:92]:
```suggestion
        startActivity(Intent(this, AboutLibrariesActivity::class.java))
```

File: app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2022 Christian Sommer
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt:2]: Wrong year

File: app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/*
+ * NOTE:
+ * The library can automatically generate a view.
+ * However, it uses material design elements which makes things a bit inconsistent within
+ * this app. As soon as the app is migrated to material design, this and related code
+ * maybe become obsolete.
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt:23]: `... may become obsolete`

File: app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+import android.os.Bundle
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.mikepenz.aboutlibraries.Libs
+import com.mikepenz.aboutlibraries.entity.Library
+import com.yacgroup.yacguide.utils.ActivityUtils
+
+class AboutLibrariesActivity : AppCompatActivity () {
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt:36]: Remove space between `AppCompatActivity` and `()`

File: app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        _activityUtils = ActivityUtils(this)
+        setTitle(R.string.open_source_libraries)
+        setContentView(R.layout.activity_about_libraries)
+        _displaycontent()
+    }
+
+    private fun _displaycontent() {
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt:48]: Use camel case `_displayContent()`

File: app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        // The resource JSON file is automatically generated
+        // by the Gradle plugin com.mikepenz.aboutlibraries.plugin
+        val jsonStr = resources.openRawResource(R.raw.aboutlibraries)
+            .bufferedReader().use  { it.readText() }
+        val libs = Libs.Builder()
+            .withJson(jsonStr)
+            .build()
+        for (lib in libs.libraries.sortedBy { it.uniqueId }) {
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt:56]:
```suggestion
        libs.libraries.sortedBy { it.uniqueId }.forEach {
            _createEntry(it)
        }
```

File: app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    private fun _createEntry(lib: Library) {
+        layoutInflater.inflate(R.layout.about_libraries_entry, null).let {
+            val content = findViewById<LinearLayout>(R.id.aboutLibrariesContent)
+            it.setOnClickListener { _activityUtils.openUrl(lib.website.orEmpty()) }
+            it.findViewById<TextView>(R.id.aboutLibrariesName).text = lib.uniqueId
+            it.findViewById<TextView>(R.id.aboutLibrariesVersion).text = lib.artifactVersion
+            val licenses= lib.licenses.joinToString(
+                separator=", ",
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt:68]: Add spaces around `=`

File: app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }
+
+    private fun _createEntry(lib: Library) {
+        layoutInflater.inflate(R.layout.about_libraries_entry, null).let {
+            val content = findViewById<LinearLayout>(R.id.aboutLibrariesContent)
+            it.setOnClickListener { _activityUtils.openUrl(lib.website.orEmpty()) }
+            it.findViewById<TextView>(R.id.aboutLibrariesName).text = lib.uniqueId
+            it.findViewById<TextView>(R.id.aboutLibrariesVersion).text = lib.artifactVersion
+            val licenses= lib.licenses.joinToString(
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt:67]: Add space before `=`

File: app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                separator=", ",
+                transform = { license -> license.name }
+            )
+            it.findViewById<TextView>(R.id.aboutLibrariesLicenses).text = licenses
+            content.addView(it)
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/AboutLibrariesActivity.kt:72]: Remove variable `content`and 
```suggestion
            findViewById<LinearLayout>(R.id.aboutLibrariesContent).addView(it)
```

File: app/src/main/res/layout/about_libraries_entry.xml [hunk 11]
```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  Copyright (C) 2022 Christian Sommer
```
Comment (paetz) [inline app/src/main/res/layout/about_libraries_entry.xml:2]: wrong year

File: app/src/main/res/layout/about_libraries_entry.xml [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+            android:layout_weight="0.2"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="10dp"
+            android:layout_marginEnd="20dp"
+            android:textAlignment="textEnd"
+            android:textSize="14sp"
+            android:textColor="?android:textColorSecondary"/>
+    </LinearLayout>
+
+    <View
```
Comment (paetz) [inline app/src/main/res/layout/about_libraries_entry.xml:64]: I believe, we have defined `ThinDivider` and `FatDivider` somewhere. You may use it here, too.

File: app/src/main/res/layout/activity_about_libraries.xml [hunk 13]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  Copyright (C) 2022 Christian Sommer
```
Comment (paetz) [inline app/src/main/res/layout/activity_about_libraries.xml:2]: wrong year

File: app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -60,6 +60,8 @@
     <string name="data_src">Datenbezugsquelle</string>
     <string name="privacy_policy">Datenschutzerklärung</string>
     <string name="license">Lizenz</string>
+    <string name="open_source_libraries">Bibliotheken und Lizenzen</string>
```
Comment (paetz) [inline app/src/main/res/values/strings.xml:63]: I would call it `Software und Lizenzen`

File: app/src/main/res/values/strings.xml [hunk 15]
```diff
@@ -60,6 +60,8 @@
...
+    <string name="open_source_libraries_details">Bibliotheken die diese App möglich machen.</string>
```
Comment (paetz) [inline app/src/main/res/values/strings.xml:64]: `Externe Software-Bibliotheken`? Or sth. with `Abhängigkeiten`?

File: app/src/main/java/com/yacgroup/yacguide/AboutActivity.kt [hunk 16]
```diff
@@ -43,6 +43,11 @@ class AboutActivity : BaseNavigationActivity() {
         _createEntry(getString(R.string.privacy_policy), callback = { _showPrivacyPolicy() })
         _createEntry(getString(R.string.source_code_and_support), getString(R.string.github_url))
         _createEntry(getString(R.string.license), getString(R.string.license_url))
+        _createEntry(
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/AboutActivity.kt:46]: Might it make sense to migrate this view to RecyclerView using `BaseViewItemAdapter`, too? If yes, you may create a ticket for that.

**Review by paetz — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (chrgernoe) [reply app/build.gradle:172]: I forgot to remove it.

File: app/src/main/java/com/yacgroup/yacguide/AboutActivity.kt (hunk 16 above)
Comment (chrgernoe) [reply app/src/main/java/com/yacgroup/yacguide/AboutActivity.kt:46]: I will have a look. Thanks for the hint.

**Review by paetz — APPROVED** [review]

**Pull request merged**